docs/006-mobile-app-architecture.adoc

```diff
@@ -0,0 +1,109 @@
+= ADR 005 - Mobile App Architecture
+Daniel Karzel <daniel@10101.finance>
+3.0, July 29, 2022: AsciiDoc article template
+:toc:
+:icons: font
+:attributes: 2023-02-02
+
+This document contains the reasoning for high level architecture and project structure decisions for the mobile app.
+
+== Decision
+
+* Application State and data storage are handled in the mobile app's native rust code.
+** Whenever data is required the app should ask the native backend for data.
+** Widget state should be kept on the presentation layer.
+* The domain model is mostly defined in rust.
+** It is recommended to keep the internal domain model used in the business logic encapsulated in the native code and expose a separate domain model on the API.
+** There must be a separate DTO model for the app's database.
+** We will likely still need some models in Dart, but they should be added for presentation and presentation controller purposes only. Examples are: a model to cache data until it is complete and can be handed over to the API to be stored, or a model that aggregates API data for display purposes.
+* We use a feature-first structure for the project files because it will scale better.
+
+== Context
+
+=== Application State and Storage
+
+The general idea is to manage the application state and business logic in the rust backend and let the Dart code focus on the presentation of the data rather than the application state management.
+Widget state necessary for the presentation layer should be handled in Dart, but state related to data and data changes should be handled in rust.
+Data storage is handled in the native rust code; the database is managed there.
+
+For simple properties that are only necessary to take decisions in Dart (no data involved) we can use the https://docs.flutter.dev/cookbook/persistence/key-value[`shared_preferences`] key-value store.
+Examples where we used the `SharedPreferences` in the POC are boolean flags for the first startup and user seed backup.
+
+General rule of thumb:
+
+* If it is structured data / relational data it should be in the database.
+* If it is a simple flag for decision making in Dart it can be stored in the `shared_preferences`.
+
+=== Data Model
+
+The data model is mostly managed in the rust code.
+It is recommended to separate the domain model used in the rust code from the model exposed to the Dart code.
+Very simple, basic domain objects can be shared between the BL and the API layer.
```

Comment:
What does `BL` stand for?

Reply:
business logic - I can remove the abbreviation